hq/app/logic/SecurityGroupDisplay.scala

```diff
@@ -0,0 +1,23 @@
+package logic
+
+import model.{ELB, Instance, SGInUse}
+
+
+object SecurityGroupDisplay {
+
+  case class ResourceIcons(instances: Int, elbs: Int, unknown: Int)
+
+  def resourceIcons(usages: List[SGInUse]): ResourceIcons = {
+
+    val instances = usages.collect {
```

Comment:
You can traverse the `usages` list at once while collecting numbers for both Instance and ELB. It's not a big deal
but something like below would work.
```
val (ins, elb) = usages.collect {
    case Instance(_) => (1,0)
    case ELB(_) => (0, 1)
}.unzip
val instances = elb.sum
val elbs = elb.sum
```

Reply:
well there is also another method i can suggest so that there is no need to call `sum` later on.

```
val (instances, elbs) = usages.foldLeft(0,0) {
  case ( (ins, elb), Instance(_)) => (ins+1, elb)
  case ( (ins, elb), ELB(_)) => (ins, elb+1)
}
```

Reply:
Grand, planning to change this to:

```
    val (instances, elbs, unknown) = usages.foldLeft(0,0,0) {
      case ( (ins, elb, unk), Instance(_) ) => (ins+1, elb, unk)
      case ( (ins, elb, unk), ELB(_) ) => (ins, elb+1, unk)
      case ( (ins, elb, unk), _ ) => (ins, elb, unk+1)
    }
```